zaza/openstack/charm_tests/ovn/tests.py

```diff
@@ -701,3 +701,134 @@ def run_tests(self):
+
+    def _cluster_status_action(self):
+        """Return Southbound and Northbound cluster status.
+
+        This function returns data as reported by "cluster-status" action
+        parsed into two dictionaries in the following order:
+        "Southbound status", "Northbound status"
+        """
+        yaml_load_err = "Status of '{}' could not be loaded as yaml:\n{}"
+        status_raw = zaza.model.run_action_on_leader("ovn-central",
+                                                     "cluster-status")
+        status_data = status_raw.data["results"]
+        # Verify expected items in the action result
+        self.assertIn("northbound-cluster", status_data)
+        self.assertIn("southbound-cluster", status_data)
+
+        try:
+            nb_status = yaml.safe_load(status_data["northbound-cluster"])
+        except yaml.YAMLError:
+            self.fail(yaml_load_err.format("northbound-cluster",
+                                           status_data["northbound-cluster"]))
+        try:
+            sb_status = yaml.safe_load(status_data["southbound-cluster"])
+        except yaml.YAMLError:
+            self.fail(yaml_load_err.format("southbound-cluster",
+                                           status_data["southbound-cluster"]))
+
+        return sb_status, nb_status
+
+    def test_cluster_status(self):
+        """Test that cluster-status action returns expected results."""
+        application = zaza.model.get_application("ovn-central")
+        sb_status, nb_status = self._cluster_status_action()
+
+        # Verify that cluster status includes "Servers" field with correct type
+        for status in (nb_status, sb_status):
+            self.assertIn("Servers", status)
+            self.assertIsInstance(status["Servers"], dict)
+
+        # Verify that units and their Server IDs are properly paired
+        expected_mapping = {}
+        for unit in application.units:
+            unit_name = unit.entity_id
+            nb_status_cmd = self.NB_CMD.format("cluster/status OVN_Northbound")
+            sb_status_cmd = self.SB_CMD.format("cluster/status OVN_Southbound")
+            nb_cluster_status = zaza.model.run_on_unit(unit_name,
+                                                       nb_status_cmd)
+            sb_cluster_status = zaza.model.run_on_unit(unit_name,
+                                                       sb_status_cmd)
+            nb_id = nb_cluster_status["Stdout"].splitlines()[0]
+            sb_id = sb_cluster_status["Stdout"].splitlines()[0]
+            expected_mapping[unit_name] = {"sb_id": sb_id, "nb_id": nb_id}
+
+        for unit_name, unit_data in expected_mapping.items():
+            sb_id = unit_data["sb_id"]
+            nb_id = unit_data["nb_id"]
+            self.assertEqual(sb_status["Servers"][sb_id]["Unit"], unit_name)
+            self.assertEqual(nb_status["Servers"][nb_id]["Unit"], unit_name)
+
+    def test_cluster_kick(self):
```

Comment:
The approach of modifying the test bundles to have a extra unit for this test is not acceptable.

Please extend this test to add/remove units when needed [0] and make sure you leave the model in a usable state for other tests to succeed when you're done.

Please also add a test that confirms that a regular remove-unit procedure leads to successful self-removal from the cluster.

0: https://github.com/openstack-charmers/zaza/blob/87e7a9c431c89ad1577106d0eda1818561f343a9/zaza/model.py#L2680